Orange/widgets/data/owpivot.py

```diff
@@ -0,0 +1,865 @@
+
+    def _initialize(self, agg_funs, val_var):
+        var_indep_funs, var_dep_funs = self.__group_aggregations(agg_funs)
+        self._create_group_tables(var_indep_funs, var_dep_funs)
+        self.__reference_aggregations(var_indep_funs, var_dep_funs)
+        self._create_pivot_tables(val_var)
+
+    def __group_aggregations(self, agg_funs):
+        auto_funcs = self.AutonomousFunctions
+        var_indep_funs = [fun for fun in agg_funs if fun in auto_funcs]
+        var_dep_funs = []
+        attrs = self._table.domain.variables + self._table.domain.metas
+        prod = product(filter_visible(attrs),
+                       [fun for fun in agg_funs if fun not in auto_funcs])
+        for var, fun in prod:
+            if self.__include_aggregation(fun, var):
+                var_dep_funs.append((var, fun))
+        return var_indep_funs, var_dep_funs
+
+    def __include_aggregation(self, fun, var):
+        return fun in self.ContVarFunctions and var.is_continuous or \
+               fun in self.DiscVarFunctions and var.is_discrete or \
+               fun in self.AnyVarFunctions
+
+    def __reference_aggregations(self, var_indep_funs, var_dep_funs):
+        self._indepen_agg_done = {}
+        self._depen_agg_done = defaultdict(dict)
+        i = 1 - int(bool(self.single_var_grouping))
+        for i, fun in enumerate(var_indep_funs, i + 1):
+            self._indepen_agg_done[fun] = i
+        for j, (var, fun) in enumerate(var_dep_funs, i + 1):
+            self._depen_agg_done[fun].update({var: j})
+
+    def _create_group_tables(self, var_indep_funs, var_dep_funs):
+        attrs = [ContinuousVariable(f"({str(fun).lower()})")
+                 for fun in var_indep_funs]
+        for var, fun in var_dep_funs:
+            name = f"{var.name} ({str(fun).lower()})"
+            if fun in self.DiscVarFunctions:
+                attrs.append(DiscreteVariable(name, var.values))
+            else:
+                attrs.append(ContinuousVariable(name))
+        args = (var_indep_funs, var_dep_funs, attrs)
+        for t, var in (("table", None), ("total_h", self._col_var),
+                       ("total_v", self._row_var), ("total", self._total_var)):
+            setattr(self._group_tables, t, self.__get_group_table(var, *args))
+
+    def __get_group_table(self, var, var_indep_funs, var_dep_funs, attrs):
+        if var is self._total_var:
+            group_tab = self._group_tables.total
+            offset = int(bool(not self.single_var_grouping))
+            leading_vars = [self._total_var]
+            combs = np.array([[0]])
+            sub_table_getter = lambda x: \
+                self._table[np.where((~np.isnan(self._row_var_col)) &
+                                     (~np.isnan(self._col_var_col)))[0]]
+        elif var is self._row_var or self.single_var_grouping:
+            group_tab = self._group_tables.total_v
+            offset = int(bool(not self.single_var_grouping))
```

Comment:
You can skip `int(bool(...))`. Operator `not` already gives a `bool` which is a subclass of `int`.